pkg/operator/controller/ingress/controller_test.go

```diff
@@ -1710,6 +1737,34 @@ func Test_computeUpdatedInfraFromService(t *testing.T) {
 			expectUpdated: true,
 			expectError:   false,
 		},
+		{
+			description:   "aws platform without DNSType set",
+			platform:      &awsPlatform,
+			ingresses:     []corev1.LoadBalancerIngress{},
+			expectUpdated: false,
+			expectError:   false,
+		},
+		{
+			description:   "aws platform with DNSType and no LB IP",
+			platform:      &awsPlatformWithDNSType,
+			ingresses:     []corev1.LoadBalancerIngress{},
+			expectUpdated: false,
+			expectError:   false,
+		},
+		{
+			description:   "aws platform with DNSType and no LB IP in infra config, service has 1 IP",
+			platform:      &awsPlatformWithDNSType,
+			ingresses:     awsIngresses,
+			expectUpdated: true,
+			expectError:   false,
+		},
+		{
+			description:   "aws platform with no change to LB IPs",
+			platform:      &awsPlatformWithLBIP,
+			ingresses:     awsIngresses,
+			expectUpdated: false,
+			expectError:   false,
+		},
```

Comment:
To be complete, you could consider adding another case like:
```
		{
			description:   "gcp platform with DNSType and LB IP",
			platform:      &gcpPlatformWithLBIP,
			ingresses:     ingressesWithMultipleIPs,
			expectUpdated: true,
			expectError:   false,
		},
```

In which both the platform and Ingress specify IPs, but it's different changes. You just can't specify multiple Ingresses, but you can change the single IP value right?